src/coroutine/memoset/mod.rs

```diff
@@ -683,8 +679,9 @@ impl<F: LurkField, Q: Query<F>> Scope<Q, LogMemo<F>, F> {
         // (insertions) before then proving itself (making use of any subquery results) and removing the now-proved
         // deferral from the MemoSet.
         for index in 0..Q::count() {
-            for key in unique_keys.get(&index).expect("unreachable") {
-                for kv in insertions.get(key).unwrap().iter() {
+            if let Some(keys) = unique_keys.get(&index) {
```

Comment:
This was actually wrong. It's entirely possible that there exist a Coroutine that is not queried in a given evaluation to be proved.